lib/upgradedb/mysql.2022091400.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * LMS version 1.11-git
+ *
+ *  (C) Copyright 2001-2016 LMS Developers
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License Version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+ *  USA.
+ *
+ */
+
+$this->BeginTrans();
+
+$this->Execute("ALTER TABLE events ADD COLUMN netnodeid int(11) DEFAULT NULL");
+$this->Execute("ALTER TABLE events ADD CONSTRAINT event_netnodeid_fkey
+    FOREIGN KEY (netnodeid) REFERENCES netnodes (id) ON DELETE SET NULL ON UPDATE CASCADE");
+
+$this->Execute("ALTER TABLE events ADD COLUMN netdevid int(11) DEFAULT NULL");
+$this->Execute("ALTER TABLE events ADD CONSTRAINT event_netdevid_fkey
+    FOREIGN KEY (netnodeid) REFERENCES netdevices (id) ON DELETE SET NULL ON UPDATE CASCADE");
+
+$this->Execute("ALTER TABLE events ADD INDEX events_netnodeid_idx (netnodeid)");
+$this->Execute("ALTER TABLE events ADD INDEX events_netdevid_idx (netdevid)");
```

Comment:
<pre><font color="#F66151"><b>[vostro3560</b></font><font color="#2A7BDE"><b> lms.github]#</b></font> grep &quot;ADD INDEX&quot; lib/upgradedb/mysql.20*
<font color="#A347BA">lib/upgradedb/mysql.2004090700.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cash <font color="#F66151"><b>ADD INDEX</b></font> (userid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005052300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cash <font color="#F66151"><b>ADD INDEX</b></font> customerid (customerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005060300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cash <font color="#F66151"><b>ADD INDEX</b></font> docid (docid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005060300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE invoicecontents <font color="#F66151"><b>ADD INDEX</b></font> docid (docid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005060300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documents <font color="#F66151"><b>ADD INDEX</b></font> cdate (cdate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005060400.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cash <font color="#F66151"><b>ADD INDEX</b></font> reference (reference)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005080300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documents <font color="#F66151"><b>ADD INDEX</b></font> numberplanid (numberplanid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005122800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE receiptcontents <font color="#F66151"><b>ADD INDEX</b></font> regid (regid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2006061200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE passwd <font color="#F66151"><b>ADD INDEX</b></font> ownerid (ownerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2006061200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE domains <font color="#F66151"><b>ADD INDEX</b></font> ownerid (ownerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2006082300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE assignments <font color="#F66151"><b>ADD INDEX</b></font> nodeid (nodeid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2006082300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE assignments <font color="#F66151"><b>ADD INDEX</b></font> customerid (customerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2006111100.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rttickets <font color="#F66151"><b>ADD INDEX</b></font> createtime (createtime)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2006111100.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rttickets <font color="#F66151"><b>ADD INDEX</b></font> customerid (customerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007011200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE nodes <font color="#F66151"><b>ADD INDEX</b></font> ownerid (ownerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007011200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE nodes <font color="#F66151"><b>ADD INDEX</b></font> ipaddr_pub (ipaddr_pub)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007022200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rtmessages <font color="#F66151"><b>ADD INDEX</b></font> ticketid (ticketid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007022600.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documents <font color="#F66151"><b>ADD INDEX</b></font> customerid (customerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007070400.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documentcontents <font color="#F66151"><b>ADD INDEX</b></font> fromdate (fromdate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007070400.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documentcontents <font color="#F66151"><b>ADD INDEX</b></font> todate (todate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007110600.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rttickets <font color="#F66151"><b>ADD INDEX</b></font> creatorid (creatorid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2007111400.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE tariffs <font color="#F66151"><b>ADD INDEX</b></font> type (type)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2008032100.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documents <font color="#F66151"><b>ADD INDEX</b></font> closed (closed)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2008080800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE customers <font color="#F66151"><b>ADD INDEX</b></font> zip (zip)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2009060200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cash <font color="#F66151"><b>ADD INDEX</b></font> importid (importid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2009060200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cash <font color="#F66151"><b>ADD INDEX</b></font> sourceid (sourceid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2009082500.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE customers <font color="#F66151"><b>ADD INDEX</b></font> name (lastname, name)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010011300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE nodeassignments <font color="#F66151"><b>ADD INDEX</b></font> assignmentid (assignmentid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010020700.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE netdevices <font color="#F66151"><b>ADD INDEX</b></font> channelid (channelid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010062200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE assignments <font color="#F66151"><b>ADD INDEX</b></font> numberplanid (numberplanid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010062800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rtattachments <font color="#F66151"><b>ADD INDEX</b></font> messageid (messageid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010062800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rtnotes <font color="#F66151"><b>ADD INDEX</b></font> (userid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010121000.php</font><font color="#2AA1B3">:</font>$this-&gt;EXECUTE(&quot;ALTER TABLE customerassignments <font color="#F66151"><b>ADD INDEX</b></font> customerid (customerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010121000.php</font><font color="#2AA1B3">:</font>$this-&gt;EXECUTE(&quot;ALTER TABLE excludedgroups <font color="#F66151"><b>ADD INDEX</b></font> customergroupid (customergroupid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010122000.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cashimport <font color="#F66151"><b>ADD INDEX</b></font> sourcefileid (sourcefileid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010122000.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cashimport <font color="#F66151"><b>ADD INDEX</b></font> customerid (customerid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2010122000.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE cashimport <font color="#F66151"><b>ADD INDEX</b></font> sourceid (sourceid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2011030700.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE promotionschemas <font color="#F66151"><b>ADD INDEX</b></font> ctariffid (ctariffid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2011082800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE netdevices <font color="#F66151"><b>ADD INDEX</b></font> location_street (location_street)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2011082800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE netdevices <font color="#F66151"><b>ADD INDEX</b></font> location_city (location_city, location_street, location_house, location_flat)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2011082800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE nodes <font color="#F66151"><b>ADD INDEX</b></font> location_city (location_city, location_street, location_house, location_flat)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2011082800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE nodes <font color="#F66151"><b>ADD INDEX</b></font> location_street (location_street)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2013031800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE pna <font color="#F66151"><b>ADD INDEX</b></font> streetid (streetid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2013031800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE pna <font color="#F66151"><b>ADD INDEX</b></font> cityid (cityid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2013031800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rtcategoryusers <font color="#F66151"><b>ADD INDEX</b></font> categoryid (categoryid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2013031800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE rtticketcategories <font color="#F66151"><b>ADD INDEX</b></font> categoryid (categoryid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2014012700.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documents <font color="#F66151"><b>ADD INDEX</b></font> reference (reference)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2014072500.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE documents <font color="#F66151"><b>ADD INDEX</b></font> fullnumber (fullnumber)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015040200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE nodes <font color="#F66151"><b>ADD INDEX</b></font> linkradiosector (linkradiosector)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015040800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE netlinks <font color="#F66151"><b>ADD INDEX</b></font> srcradiosector (srcradiosector)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015040800.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE netlinks <font color="#F66151"><b>ADD INDEX</b></font> dstradiosector (dstradiosector)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015072300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE voipaccounts <font color="#F66151"><b>ADD INDEX</b></font> location_city (location_city, location_street, location_house, location_flat)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015072300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE voipaccounts <font color="#F66151"><b>ADD INDEX</b></font> location_street (location_street)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015112000.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;ALTER TABLE events <font color="#F66151"><b>ADD INDEX</b></font> nodeid (nodeid)&quot;);
<font color="#F66151"><b>[vostro3560</b></font><font color="#2A7BDE"><b> lms.github]#</b></font> grep &quot;CREATE INDEX&quot; lib/upgradedb/mysql.20*
<font color="#A347BA">lib/upgradedb/mysql.2005012600.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> events_date_idx ON events(date)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005020600.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> invoiceid ON cash(invoiceid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005022500.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> tariffid ON assignments(tariffid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005030200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> netdev ON nodes(netdev)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005030200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> queueid ON rttickets(queueid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005030200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> time ON cash(time)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005030200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> cdate ON invoices(cdate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005030200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> invoiceid ON invoicecontents(invoiceid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2005030200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> hash ON cashimport(hash)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2012042300.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> nodesessionid ON stats(nodesessionid)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2015042700.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> authtype ON nodes(authtype)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2018022200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> pna_fromnumber_idx ON pna (fromnumber)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2018022200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> pna_tonumber_idx ON pna (tonumber)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2019041200.php</font><font color="#2AA1B3">:</font>$this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> cash_linktechnology_idx ON cash (linktechnology)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2021050701.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> customerassignments_startdate_idx ON customerassignments (startdate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2021050701.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> customerassignments_enddate_idx ON customerassignments (enddate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022061300.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_cdate_idx ON voip_cdr (cdate)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022070400.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_caller_idx ON voip_cdr (caller)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022070400.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_callee_idx ON voip_cdr (callee)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022082600.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_subtype_idx ON voip_cdr (subtype)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022082601.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_call_start_time_idx ON voip_cdr (call_start_time)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022090800.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_direction_idx ON voip_cdr (direction)&quot;);
<font color="#A347BA">lib/upgradedb/mysql.2022090800.php</font><font color="#2AA1B3">:</font>    $this-&gt;Execute(&quot;<font color="#F66151"><b>CREATE INDEX</b></font> voip_cdr_incremental_idx ON voip_cdr (incremental)&quot;);
<font color="#F66151"><b>[vostro3560</b></font><font color="#2A7BDE"><b> lms.github]#</b></font> 
</pre>

Reply:
https://github.com/chilek/lms/pull/2273/commits/6b4aa36f58eb43babfe3c18eb2f60d47e69db7ee